### DISCUSSION TITLE
Suggestion

### DISCUSSION THREAD

**Discussion by DevX-Cipher** [dotnet/msbuild#7343] [Ideas]
hi i was wondering if this can be modified to be able to build .sln without know the name and just the extension .sln only

**Comment by danmoseley** [comment]
Hello, could you give an example of what you would type?

**Comment by DevX-Cipher** [comment]
MsBuild.exe %USERPROFILE%\Desktop\Test\    .sln /property:Configuration=Debug

so you don't have to include the file name

**Comment by teo-tsirpanis** [comment]
@modz2014 if a directory has only one solution file you can `cd` into it and run MSBuild without specifying the path.

**Reply by DevX-Cipher** [reply]
Yer I realised that now

**Comment by DevX-Cipher** [comment]
wont build unless i add the sln file name 

i want to be able to use msbuild with adding the filename.sln thats why im asking if this can be modified if you guys dont want to do it can you point me in the right direction

**Comment by teo-tsirpanis** [comment]
Why doesn't it work? What kind of errors does it show?

**Comment by DevX-Cipher** [comment]
this is another example if the first one doesnt make sense

>msbuild sln /property:Configuration=Debug
Microsoft (R) Build Engine version 17.0.0+c9eb9dd64 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

MSBUILD : error MSB1009: Project file does not exist.
Switch: sln

**Comment by teo-tsirpanis** [comment]
Try `cd`ing into your solution's directory and run `msbuild /property:Configuration=Debug`, without the `sln`.

**Comment by DevX-Cipher** [comment]
im the directory of the file 

```
msbuild  /property:Configuration=Debug
Microsoft (R) Build Engine version 17.0.0+c9eb9dd64 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

MSBUILD : error MSB1011: Specify which project or solution file to use because this folder contains more than one project or solution file.
```

there is only one .sln file and .csproj

**Comment by teo-tsirpanis** [comment]
If the directory contains many solutions MSBuild will naturally not be able to find out which you are talking about, and you inevitably have to specify it yourself. If it has one solution and many projects, I suggest you put them in separate directories.

**Comment by DevX-Cipher** [comment]
ok thanks

**Comment by danmoseley** [comment]
I believe VS, when asked to open a project file, will instead open a solution if there is one in the same directory or immediately above that contains the project. MSBuild does not do that - which I think is good but possibly confusing here.

**Comment by DevX-Cipher** [comment]
Yes I think so too also question is can I discuss about a new GUI window for forms there getting old now windows 7 style is there anything that you know of I can change this ect

**Comment by DevX-Cipher** [comment]
Where would I find dotnet frame works Dev packs installed to because the location doesn't so when installing them
